backend/app/src/test/kotlin/io/tolgee/api/v2/controllers/v2ProjectsController/ProjectsControllerEditTest.kt

```diff
@@ -72,4 +74,36 @@ class ProjectsControllerEditTest : AuthorizedControllerTest() {
       node("CUSTOM_VALIDATION.namespaces_cannot_be_disabled_when_namespace_exists").isNotNull
     }
   }
+
+  @Test
+  fun `can use the same slug as deleted project has`() {
+    val base = dbPopulator.createBase()
+
+    executeInNewTransaction {
+      val deleted = dbPopulator.createBase()
+      deleted.project.deletedAt = currentDateProvider.date
+      deleted.project.slug = "new-slug-2"
+      projectService.save(deleted.project)
+    }
+    val newSlug = "new-slug-2"
```

Comment:
```suggestion
    val slugToBeReused = "new-slug-2"
```

And I would put it before the first occurence and use it instead of the copy-pasted string to keep it semantically accurate.